packages/block-library/src/details/edit.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	InnerBlocks,
+	RichText,
+	InspectorControls,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { ToggleControl } from '@wordpress/components';
+
+export default ( {
+	attributes,
+	className,
+	clientId,
+	isSelected,
+	setAttributes,
+} ) => {
+	const innerBlockSelected = useSelect(
+		( select ) =>
+			select( 'core/block-editor' ).hasSelectedInnerBlock( clientId ),
+		[ clientId ]
+	);
+
+	const showInnerBlocks =
+		attributes.initialOpen || isSelected || innerBlockSelected;
+
+	const classes = classnames(
+		{
+			'is-open': showInnerBlocks,
+		},
+		className
+	);
+
+	/* You may be wondering why we don't just use a <details> element here.
+	 * The problem we are trying to solve is that a <summary> is basically a
+	 * button, and when it has focus, it eats the space key.
+	 *
+	 * That's a problem if you want to use a <RichText> component inside your
+	 * <summary>. Each time you press space, it toggles the rest of the
+	 * <details>, and it doesn't even add a space to your <RichText>.
+	 *
+	 * Then there's the fact that the space exists for a11y reasons. If you
+	 * catch the event and preventDefault() then you've remove the way for
+	 * keyboard users to toggle the <details>.
+	 */
```

Comment:
Come to think of it, maybe we _can_ use a `details` + `summary` element in the editor after all. Why not just catch the event and `preventDefault()` in the editor? Combine that with the current expand-on-focus behavior and nothing would change except we could use the actual HTML elements and avoid the CSS simulations, right? We might also have to set a different value for the HTML `role` attribute if using the `RichText` component doesn't already do that.

We already can't click-to-toggle the simulated `details` in the editor, so why not just go ahead and use a real `details`?

Reply:
It's worth noting that a "browse mode" as proposed in #23328 would allow the user to interact with the `details` the same as they would on the front-end.

Reply:
I started with `summary` and I ran into issues with `RichText` getting the space key event when I `preventDefault()`. Even if I stopped the space key event on `summary`, the event wasn't getting to RichText for some reason so mytextwouldruntogether.

I'll spend some more time with it though, in case I missed something.

Reply:
I made some progress on `<summary>` with @vcanales's help and now it works in Chrome and Safari. I'll see what I can do about Firefox

Reply:
@vcanales and I got `details` and `summary` to in the edit view in Chrome and Safari by putting `RichText` as a child of `summary`, however Firefox seems to have an issue with `contenteditable` inside `summary` that prevents it from seeing the space key. I've created [a simple test on JSFiddle](https://jsfiddle.net/g12ng/xytha5sb/) showing the issue. This is probably due to its' role as Button. 

I can't update the value in the Edit component because I don't know the current caret position, so the only other way forward I see is to add some way to trigger a keyboard event in `RichText`.

Reply:
Hmmm... this is a tough one. Pinging @ellatrix for technical help.